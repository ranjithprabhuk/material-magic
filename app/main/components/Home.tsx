
import * as React from 'react';
import {RouteComponentProps} from 'react-router';

class Home extends React.Component<RouteComponentProps<object>, object> {
  public render(): React.ReactElement<object> {
    return (
        <div>
          <h1>Welcome to React+Redux+Typescript starter</h1>
        </div>
    );
  }
}

export default Home;
