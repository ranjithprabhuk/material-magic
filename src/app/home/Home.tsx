import * as React from 'react';
import { withStyles } from '@material-ui/core/styles';
import { Grid } from '@material-ui/core';
import { homeStyles } from './home.styles';
import { IHomeProps, IHomeState } from './IHome';
import { StatsWidget, CollapsiblePanel } from '../components';
import StarIcon from '@material-ui/icons/Star';
import InboxIcon from '@material-ui/icons/Inbox';
import { defineColours, ThemeConfig } from '../../theme';

class Home extends React.Component<IHomeProps, IHomeState> {
  public render(): React.ReactElement<Home> {
    const { classes } = this.props;

    return (
      <div className={classes.root}>
        <Grid container spacing={8}>
          <Grid item xs={12} sm={6} md={3}>
            <StatsWidget
              component={<StarIcon style={{ fontSize: 50 }} />}
              value='32.5 C'
              title='Temperature'
              textColor={defineColours(ThemeConfig.primaryColor, true)}
              backgroundColor={defineColours(ThemeConfig.primaryColor)}
            />
          </Grid>
          <Grid item xs={12} sm={6} md={3}>
            <StatsWidget
              component={<InboxIcon style={{ fontSize: 50 }} />}
              value='13486'
              title='Visibility'
              color='secondary'
              backgroundColor={defineColours(ThemeConfig.secondaryColor)}
            />
          </Grid>
          <Grid item xs={12} sm={6} md={3}>
            <StatsWidget
              component={<StarIcon style={{ fontSize: 50 }} />}
              value='87%'
              title='Humidity'
              textColor={defineColours(ThemeConfig.warningColor, true)}
              backgroundColor={defineColours(ThemeConfig.warningColor)}
            />
          </Grid>
          <Grid item xs={12} sm={6} md={3}>
            <StatsWidget
              component={<StarIcon style={{ fontSize: 50 }} />}
              value='2.1 m/s'
              title='Wind Speed'
              textColor={defineColours(ThemeConfig.dangerColor, true)}
              backgroundColor={defineColours(ThemeConfig.dangerColor)}
            />
          </Grid>
        </Grid>
        <Grid container spacing={8}>
          <Grid item xs={12} sm={6} md={3}>
            <CollapsiblePanel />
          </Grid>
        </Grid>
      </div>
    );
  }
}

export default withStyles(homeStyles, {withTheme: true})(Home);
