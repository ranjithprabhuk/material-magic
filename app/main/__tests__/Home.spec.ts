import * as React from 'react';
import Home from '../components/Home';
import {createRenderer, ShallowRenderer} from 'react-test-renderer/shallow';


describe('Home', () => {
  let renderer:ShallowRenderer;
  beforeEach(() => {
    renderer = createRenderer();
  });

  afterEach(() => {
    // empty
  });

  it('should render without crashing', () => {
    const home = renderer.render( React.createElement(Home) );
    expect(home).toMatchSnapshot();
  });
});
