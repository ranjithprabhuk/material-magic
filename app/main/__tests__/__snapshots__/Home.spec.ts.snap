// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home should render without crashing 1`] = `
<div>
  <h1>
    Welcome to React+Redux+Typescript starter
  </h1>
</div>
`;
